cnxpublishing/subscribers.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+import logging
+
+import psycopg2
+from cnxarchive.scripts import export_epub
+from pyramid.events import subscriber
+from pyramid.threadlocal import get_current_registry
+
+
+from . import events
+from .bake import remove_baked, bake
+from .db import (
+    set_post_publications_state,
+    update_module_state,
+    with_db_cursor,
+)
+
+
+logger = logging.getLogger('cnxpublishing')
+
+
+@subscriber(events.PostPublicationEvent)
+@with_db_cursor
+def post_publication_processing(event, cursor):
+    """Process post-publication events coming out of the database."""
+    module_ident, ident_hash = event.module_ident, event.ident_hash
+    logger.debug('Processing module_ident={} ident_hash={}'.format(
+        module_ident, ident_hash))
+    set_post_publications_state(cursor, module_ident, 'Processing')
+    # TODO commit state change
```

Comment:
This is critical - some of our current baking can take a long time, and we need that visible to other processes. Especially since we're still one-job-at-a-time on this. The current post_processing daemon does make stateid == 6 (Processing) visible in the db. I presume the change in transaction handling is to avoid exposing the deletion of existing baked content until success?